### PR TITLE
libpod: avoid polling container status

### PR DESCRIPTION
**Pull request by giuseppe**
use the inotify backend to be notified on the container exit instead
of polling continuosly the runtime.  Polling the runtime slowns
significantly down the podman execution time for short lived
processes:

$ time bin/podman run --rm -ti fedora true

real	0m0.324s
user	0m0.088s
sys	0m0.064s

from:

$ time podman run --rm -ti fedora true

real	0m4.199s
user	0m5.339s
sys	0m0.344s

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: libpod/container_api.go [hunk 1]
```diff
@@ -524,6 +524,14 @@ func (c *Container) WaitWithInterval(waitTimeout time.Duration) (int32, error) {
...
 	}
+
+	exitFile := c.exitFilePath()
+	chWait := make(chan error)
+	defer close(chWait)
+	if _, err := WaitForFile(exitFile, chWait, 0); err != nil {
+		logrus.Errorf("Error waiting for file %s: %v", exitFile, err)
+	}
+
 	err := wait.PollImmediateInfinite(waitTimeout,
```
Comment (mheon) [inline libpod/container_api.go]: Do we still need this, if we're doing the waiting for the file above? I'd just as soon drop the polling wait and convert entirely to `WaitForFile()`

Comment (giuseppe) [reply libpod/container_api.go]: tests are failing as they get stuck in WaitForFile.  I was just rewriting it to use both (in a select block).  What do you think?

Comment (mheon) [reply libpod/container_api.go]: Doesn't WaitForFile already do a poll alongside the inotify? I don't think we need to keep another one here. Might want to let you set the interval on the polling in WaitForFile though.

Failures are probably the file never appearing due to `--rm` - we'll need a timeout on this.

Comment (giuseppe) [reply libpod/container_api.go]: fixed in the last version I've pushed

**Comment by mheon** [comment]
LGTM on my end - nice work @giuseppe

**Comment by rhatdan** [comment]
/lgtm
Awesome job @giuseppe

**Pull request merged**